src/relax/transform/utils.h

```diff
@@ -47,6 +84,20 @@ bool IsNestedTensor(const StructInfo& sinfo);
  */
 bool IsNestedTensor(const Expr& expr);
 
+/*!
+ * \brief Check if the given StructInfo is a nested tensor of floating point dtype.
+ * \param sinfo The StructInfo to be checked.
+ * \return true if the given StructInfo is a nested tensor of floating point dtype.
+ */
+bool IsNestedFloatTensor(const StructInfo& sinfo);
+
+/*!
+ * \brief Check if the given expr is a nested tensor of floating point dtype.
+ * \param expr The expr to be checked.
+ * \return true if the given expr is a nested tensor of floating point dtype.
+ */
+bool IsNestedFloatTensor(const Expr& expr);
+
```

Comment:
Thanks @Ubospica for supporting IsNestedTensorConditioned. Since each pass may have its own specific condition, it is very likely that IsNestedFloatTensor has rare use. Compared with “adding a util function for each pass”, we more often encourage the passes to directly use IsNestedTensorConditioned - providing a lambda on the caller side is very easy. So here we don’t necessarily need IsNestedFloatTensor.

BTW you can have a version `bool IsNestedTensorConditioned(const Expr& expr, FType f_condition)` for your convenience.